service/raft/raft_group0_client.cc

```diff
@@ -251,12 +251,22 @@ future<group0_guard> raft_group0_client::start_operation(seastar::abort_source*
     auto [upgrade_lock_holder, upgrade_state] = co_await get_group0_upgrade_state();
     switch (upgrade_state) {
         case group0_upgrade_state::use_post_raft_procedures: {
-            auto operation_holder = co_await get_units(_operation_mutex, 1);
+            auto operation_holder = co_await std::invoke([this, as] {
+                if (!as) {
+                    return get_units(_operation_mutex, 1);
+                }
+                return get_units(_operation_mutex, 1, *as);
+            });
             co_await _raft_gr.group0_with_timeouts().read_barrier(as, timeout);
```

Comment:
We should eventually also fix the semaphore waits to respect the passed-in `raft_timeout` as well, but that's out of scope of this PR and not urgent.

Please open a follow-up issue for that.

Reply:
Created the #19867 for that.